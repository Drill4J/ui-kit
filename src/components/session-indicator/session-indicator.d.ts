interface Props {
    className?: string;
    active?: boolean;
}
export declare const SessionIndicator: {
    ({ className, active }: Props): JSX.Element;
    displayName: string;
    __docgenInfo: {
        description: string;
        displayName: string;
        props: {
            className: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            active: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
        };
    };
};
export {};
