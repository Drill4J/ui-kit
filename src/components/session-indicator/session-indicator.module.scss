@import '../../style-guide/index';

.session-indicator {
  width: 32px;
  height: 32px;

  &__circle-path {
    fill: map-get($green, default);
    transform-origin: 50% 50%;
    animation: rotation 2s linear infinite;
  }

  &__eye-path {
    stroke: map-get($green, default);
    stroke-width: 1.3;

    &--active {
      stroke: none;
      fill: map-get($red, default);
      animation: blinker 1.5s infinite cubic-bezier(1, 0, 0, 1);
    }
  }
}

@keyframes rotation {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes blinker {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}
