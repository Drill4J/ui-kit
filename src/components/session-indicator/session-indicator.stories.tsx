import React from 'react';
import { ComponentMeta } from '@storybook/react';
import 'twin.macro';

import { SessionIndicator } from './session-indicator';

export default {
  title: 'SessionIndicator',
  component: SessionIndicator,
} as ComponentMeta<typeof SessionIndicator>;

const Template = (args) => <SessionIndicator {...args} />;
export const Default = Template.bind({});

export const Active = Template.bind({});
Active.args = { active: true };
