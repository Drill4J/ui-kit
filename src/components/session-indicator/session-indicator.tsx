import * as React from 'react';
import { BEM, tag } from '@redneckz/react-bem-helper';

import { path as circlePath } from './circle.path';
import { path as eyePath } from './eye.path';

import styles from './session-indicator.module.scss';

interface Props {
  className?: string;
  active?: boolean;
}

const sessionIndicator = BEM(styles);

export const SessionIndicator = sessionIndicator(({
  className,
  active,
}: Props) => (
  <svg className={className} viewBox="0 0 32 32">
    <g fillRule="nonzero" stroke="none" strokeWidth="1" fill="none">
      <CirclePath d={circlePath} />
      {active
        ? <EyePath d={eyePath} active />
        : <EyePath d={eyePath} />}
    </g>
  </svg>
));

const CirclePath = sessionIndicator.circlePath(tag('path')({ d: '' } as { d: string }));
const EyePath = sessionIndicator.eyePath(tag('path')({ d: '' } as { d: string; active?: boolean }));
