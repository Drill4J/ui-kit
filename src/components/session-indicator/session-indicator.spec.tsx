import React from 'react';
import renderer from 'react-test-renderer';

import { SessionIndicator } from './session-indicator';

describe('SessionIndicator', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<SessionIndicator active />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
