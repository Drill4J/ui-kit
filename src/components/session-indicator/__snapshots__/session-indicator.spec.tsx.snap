// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionIndicator should match snapshot 1`] = `
<svg
  className="session-indicator session-indicator--active"
  viewBox="0 0 32 32"
>
  <g
    fill="none"
    fillRule="nonzero"
    stroke="none"
    strokeWidth="1"
  >
    <path
      className="session-indicator__circle-path"
      d="M15.993 32A15.999 15.999 0 0 1 7.061 2.73 15.998 15.998 0 0 1 30.83 22.004a.709.709 0 1 1-1.312-.531c2.881-7.063-.143-15.154-6.95-18.595C15.763-.562 7.455 1.8 3.477 8.308c-3.978 6.507-2.293 14.98 3.873 19.47 6.166 4.489 14.745 3.491 19.717-2.293a.709.709 0 1 1 1.077.922A15.992 15.992 0 0 1 15.993 32z"
    />
    <path
      className="session-indicator__eye-path session-indicator__eye-path--active"
      d="M15.993 11.643a4.336 4.336 0 0 0-3.075 1.274 4.336 4.336 0 0 0-1.275 3.076c0 1.201.487 2.289 1.275 3.076a4.342 4.342 0 0 0 6.15-.002 4.343 4.343 0 0 0 .001-6.15 4.336 4.336 0 0 0-3.076-1.274z"
    />
  </g>
</svg>
`;
